RoboJackets/igvc-software
# IMU Improvements and Research Document

# Description

The IMU has been having problems right after sensors are launched, suspicion was placed on the magnetometer but the cause appears to be the IMU's filtering. This PR gives more tools to control the values and observe them, and adds a research document explaining results.

This PR does the following:
- Requests magnetometer data and publishes it in addition to IMU message.
- Adds gyro calibration to launchfiles, only calibrates when explicitly requested.
- Adds research document detailing information about IMU problems.

Fixes #633 

# Testing steps (If relevant)
## Test Case 1
1. Connect the top IMU to the robot and launch `imu_top.launch`. Do not move the robot during the next steps.
2. Launch `rqt_plot` and add `/magnetometer/mag` to visualization. It should be noisy but stable.
3. Add `/magnetometer/orientation` to visualization. Observe for at least one minute. It should gradually converge to a value or be stable depending on when it's launched.
4. Relaunch `imu_top.launch`, repeat 3 multiple times. They should converge to same value.

Expectation: The IMU publishes magnetic field in addition to the IMU data, gyro doesn't recalibrate unless requested and convergence is consistent.

# Self Checklist
